samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -951,30 +971,55 @@ static bool event_handler(const struct event_header *eh)
 	}
 
 	if (is_wake_up_event(eh)) {
-		if (atomic_cas(&state, STATE_SUSPENDED, STATE_FETCHING)) {
+		k_spinlock_key_t key = k_spin_lock(&state.lock);
+		if (state.state == STATE_SUSPENDED) {
+			gpio_pin_disable_callback(gpio_dev,
```

Comment:
Shouldn't you only disable callback when going to FETCHING state after wake up? If there is no peer connected, you will go to IDLE with disabled motion interrupt.

Reply:
This is intentional; I re-enable the callback when connection happens while keeping state idle.
The justification is that data from the sensor are ignored anyway as there is no place to send when we are disconnected.

There is however one thing that I forgot about and I need to check it. When we reconnect and sample the data for the first time we will get everything that sensor accumulated (since we are not reading out the data). Guess the best thing to do is to ignore first sample after connection happens.

Reply:
I agree, there's no use in sampling the sensor when no one is connected.

Regarding the accumulation of data - I guess sensor has some timeout after which it zeroes the values in registers. But for very quick reconnection probably you could encounter the accumulated value. I cannot see any timeout value in doc. If you see this happening, you can drop the first motion read. If the time from connection to second motion burst read would be too long, you could just read `OPTICAL_REG_MOTION` register to ignore the motion values. See decription of register `0x02`:
> if Delta_X_L, Delta_X_H, Delta_Y_L and Delta_Y_H registers are not read before the motion register is read for the second time, the data in Delta_X_L, Delta_X_H, Delta_Y_L and Delta_Y_H will be lost.

Reply:
thanks. I will add logic to sample out such data but that can mean we will not be able to disable interrupt in case we are not connected (I prefer it to dropping the first sample in case we are connected).

Reply:
Actually, I never observed a huge peak in motion value sampled. Probably the timeout is shorter than any possible reconnection time. What about disabling interrupt in disconnected mode and not dropping the first sample (or at least checking it + limiting)?

Reply:
I have left this part of logic as it was so far, meaning that if we are fetching but not sending we will retry every x ms just to clear the sensor backlog. It should be good enough for disconnected state and in connected we will have rapid response.